packages/flutter/lib/src/material/pickers/date_picker_dialog.dart

```diff
@@ -0,0 +1,480 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import '../button_bar.dart';
+import '../button_theme.dart';
+import '../color_scheme.dart';
+import '../debug.dart';
+import '../dialog.dart';
+import '../flat_button.dart';
+import '../icons.dart';
+import '../material_localizations.dart';
+import '../text_theme.dart';
+import '../theme.dart';
+
+import 'calendar_date_picker.dart';
+import 'date_picker_common.dart';
+import 'date_picker_header.dart';
+import 'date_utils.dart' as utils;
+import 'input_date_picker.dart';
+
+const Size _calendarPortraitDialogSize = Size(330.0, 518.0);
+const Size _calendarLandscapeDialogSize = Size(496.0, 346.0);
+const Size _inputPortraitDialogSize = Size(330.0, 270.0);
+const Size _inputLandscapeDialogSize = Size(496, 160.0);
+const Duration _dialogSizeAnimationDuration = Duration(milliseconds: 200);
+
+/// Shows a dialog containing a Material Design date picker.
+///
+/// The returned [Future] resolves to the date selected by the user when the
+/// user confirms the dialog. If the user cancels the dialog, null is returned.
+///
+/// The [initialDate] is the initially shown and selected date in the picker.
```

Comment:
This sentence is a little hard to parse.

Maybe something like: When the date picker is first displayed, [initialDate] is selected.